src/components/search/SearchResults.jsx

```diff
@@ -78,63 +81,75 @@ const SearchResults = ({
+          {isSearchStalled && (
+            <Skeleton className="h2 d-block mb-3" width={240} />
           )}
-          {(!isSearchStalled && nbHits > 0) && (contentType === undefined) && (
-            <button
-              onClick={clickHandler}
-              className="show-all-link btn btn-link muted-link inline-link d-inline-block pl-0 pr-4 px-xl-0"
-              type="button"
-            >
-              {linkText}
-            </button>
+          {!isSearchStalled && nbHits > 0 && (
+            <>{resultsHeading}</>
           )}
-        </div>
-        {isSearchStalled && (
-          <>
-            <Skeleton className="lead mb-4" width={160} />
-            <Row>
-              {[...Array(getNoOfResultsFromTitle(title)).keys()].map(resultNum => (
-                <div key={resultNum} className="skeleton-course-card">
-                  {getSkeletonCardFromTitle(title)}
-                </div>
-              ))}
-            </Row>
-          </>
-        )}
-        {!isSearchStalled && nbHits > 0 && (
-          <>
-            <Hits hitComponent={hitComponent} />
-            {(contentType !== undefined) && (
-              <div className="d-flex justify-content-center">
-                <SearchPagination defaultRefinement={page} />
-              </div>
-            )}
-          </>
+        </h2>
+        {(!isSearchStalled && nbHits > 0) && (contentType !== undefined) && (
+          <SearchPagination
+            defaultRefinement={page}
+            maxPagesDisplayed={5}
+          />
         )}
-        {!isSearchStalled && nbHits === 0 && getContentTypeFromTitle(title) !== CONTENT_TYPE_PATHWAY && (
-          <SearchNoResults title={title} />
+        {(!isSearchStalled && nbHits > 0) && (contentType === undefined) && (
+          <button
+            onClick={clickHandler}
+            className="show-all-link btn btn-link muted-link inline-link d-inline-block pl-0 pr-4 px-xl-0"
+            type="button"
+          >
+            {linkText}
+          </button>
         )}
-        {!isSearchStalled && isDefinedAndNotNull(error) && showMessage(contentType, title) && (
-          <SearchError title={title} />
-        )}
-      </>
+      </div>
+      {isSearchStalled && (
+        <>
+          <Skeleton className="lead mb-4" width={160} />
+          <CardGrid
+            columnSizes={{
+              xs: 12,
+              md: 6,
+              lg: 4,
+              xl: 3,
+            }}
+          >
+            {[...Array(getNoOfResultsFromTitle(title)).keys()].map(resultNum => <SkeletonCard data-testid="skeleton-card" key={resultNum} />)}
+          </CardGrid>
+        </>
+      )}
+      {!isSearchStalled && nbHits > 0 && (
+        <>
+          <CardGrid
+            columnSizes={{
+              xs: 12,
+              md: 6,
+              lg: 4,
+              xl: 3,
+            }}
+          >
+            {hits?.map((hit) => <HitComponent key={uuidv4()} hit={hit} />)}
```

Comment:
This is a better way to generate keys then the array index like I've been doing. Will keep that in mind in the future.